xilem_core/src/sequence.rs

```diff
@@ -64,6 +64,59 @@ impl<T> Default for AppendVec<T> {
     }
 }
 
+/// The number of elements a [`ViewSequence`] can have.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Count {
+    /// This sequence is known to have no elements.
+    Zero,
+    /// This sequence is known to have exactly one element.
+    One,
+    // TODO: AtMostOne - useful for Option (and e.g. `SizedBox`?).
+    /// This sequence may have any number of elements.
+    Many,
+    /// The number of elements this sequence has is not statically known.
+    ///
+    /// This is used for [`AnyView`](crate::AnyView).
+    Unknown,
+}
```

Comment:
Maybe we can if known also specify the exact count (if possible), or something like `AtLeast(usize)` or `AtMost(usize)`, but I haven't carefully thought about this yet, just some food for thought.

Reply:
Yeah that's what I fear too, I guess when there's a real use-case for that we can still come back to this.